crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -340,29 +341,33 @@ impl CumulocityConverter {
 
     pub async fn process_health_status_message(
         &mut self,
+        entity: &EntityTopicId,
         message: &Message,
     ) -> Result<Vec<Message>, ConversionError> {
         let mut mqtt_messages: Vec<Message> = Vec::new();
 
         // Send the init messages
-        if check_tedge_agent_status(message)? {
-            create_tedge_agent_supported_ops(self.ops_dir.clone()).await?;
+        // TODO: use entity store to check service statuses
+        if is_message_tedge_agent_up(message)? {
+            create_tedge_agent_supported_ops(&self.ops_dir).await?;
             mqtt_messages.push(create_get_software_list_message()?);
         }
 
         // When there is some messages to be sent on behalf of a child device,
         // this child device must be declared first, if not done yet
-        let topic_split: Vec<&str> = message.topic.name.split('/').collect();
-        if topic_split.len() == 4 {
-            let child_id = topic_split[2];
-            add_external_device_registration_message(
-                child_id.to_string(),
-                &mut self.children,
-                &mut mqtt_messages,
-            );
+        if let Some(parent) = entity.default_parent_identifier() {
+            if parent.is_default_child_device() {
+                let child_id = parent.default_device_name().unwrap();
+                add_external_device_registration_message(
+                    child_id.to_string(),
+                    &mut self.children,
+                    &mut mqtt_messages,
+                );
+            }
```

Comment:
Beware, this logic is being re-visited by https://github.com/thin-edge/thin-edge.io/pull/2266.

Reply:
This can be fully removed. @albinsuresh  can you confirm?